Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-page-titles.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { createContext, useCallback, useContext, useMemo } from 'react';
+
+import { IHaveChildrenProps } from '../components/common/Props';
+
+interface IPageTitleParamsContext {
+    actions: {
+        setPageTitle: (title: string | undefined) => void;
+    };
+}
+
+type IPageTitleProviderProps = IHaveChildrenProps
+
+const PageTitleContext = createContext<IPageTitleParamsContext>({} as IPageTitleParamsContext);
+
+const PageTitleProvider = ({ children }: IPageTitleProviderProps) => {
```

Comment:
Again, this is not a `PageTitle`, but a `PageWithTitle`, or `TitledPage`, or something that says "This is a page with title".
`PageTitle` means "The title of the page".